File: leetcode/leetcode_cpp/kids-with-the-greatest-number-of-candies.cpp
/*
https://leetcode.com/problems/kids-with-the-greatest-number-of-candies
*/
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        vector<bool> res(candies.size());
        int m = 0;
        for (auto candy : candies) m = max(m, candy);
        for (int i=0; i<candies.size(); ++i) {
            res[i] = candies[i] + extraCandies >= m;
        }
        return res;
    }
};

File: leetcode/leetcode_py/kids-with-the-greatest-number-of-candies.py
'''
https://leetcode.com/problems/kids-with-the-greatest-number-of-candies
'''
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        res = []
        m = 0
        for candy in candies:
            m = max(m, candy)
        for candy in candies:
            res.append(m <= candy + extraCandies)                
        return res
        